products_analysis.py

```diff
@@ -0,0 +1,32 @@
+from fractions import Fraction
+from fetch_products import FetchProducts
+from constant import Constant
+
+
+class ProductsAnalysis:
+
+    @staticmethod
+    def max_discounted_products_ratio_dict():
+        discounted_products_ratio = []
+
+        for record in FetchProducts.records:
+            discount_ratio = Fraction(int(record[Constant.OLD_PRICE]), int(record[Constant.NEW_PRICE]))
+            if discount_ratio > 1:
+                discounted_products_ratio.append((record[Constant.BRAND], str(discount_ratio)))
+
+        return dict(discounted_products_ratio)
+
+    @staticmethod
+    def brand_wise_max_min_prices(find_what, brands_price_dict):
```

Comment:
Let's apply Single Responsibility Principal here. As we are calling this method min and max at two different places.